Written by: Pierre Lejeune <superheron@gmail.com>
Upstream: no, adapted for dotnet-sdk:8.0 from PR 4043
Reason: Fix versions calculcations when building in 2026+

---
 .../0003-Fix-assembly-version-calculation-in-2026.patch         | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0003-Fix-assembly-version-calculation-in-2026.patch b/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0003-Fix-assembly-version-calculation-in-2026.patch
index 3d3026f30..5b7714144 100644
--- a/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0003-Fix-assembly-version-calculation-in-2026.patch
+++ b/src/source-build-externals/patches/azure-activedirectory-identitymodel-extensions-for-dotnet/0003-Fix-assembly-version-calculation-in-2026.patch
@@ -23,7 +23,7 @@ index c5117975..e71b6c45 100755

  assemblyVersion=$(sed -n 's/.*<assemblyVersion>\([^<]*\)<.*/\1/p' ${scriptroot}/buildConfiguration.xml)
 -assemblyFileVersion="$assemblyVersion.${dateTimeStamp::$((${#dateTimeStamp} - 6))}" # Trim minutes/seconds
-+assemblyFileVersion="$assemblyVersion.$revision"
++assemblyFileVersion="$assemblyVersion.$((61232 + ${dateTimeStamp:0:1} * 416 + ${dateTimeStamp:1:2} * 32 + ${dateTimeStamp:3:2}))"
  assemblyInformationalVersion="$assemblyVersion.$dateTimeStamp"

  echo "assemblyVersion: $assemblyVersion"
--
2.52.0
